### PR TITLE
Make it so the upgrade step of a test never runs in parallel

### PR DESCRIPTION
**Pull request by krousey**
This is mainly a QOL (Quality of Logging) improvement for the parallel upgrade/downgrade suites.

I don't think it was ever really intended for the upgrade/downgrade step to run with multiple ginkgo parallel nodes. And as it is currently designed, they can only run in 1 ginkgo test suite, so the parallelism for this step is useless. Also it clogs up the logging with multiple nodes acting at the same time.

The parallelism was being set at the top level kube_test invocation and being propagated down through env vars. So this is just a quick loop to skip over those specific env vars.

### CODE REVIEW

**Comment by krousey** [comment]
Could help debug https://github.com/kubernetes/kubernetes/issues/56879

/assign @BenTheElder

**Comment by BenTheElder** [comment]
Let's fast track to get these solved for the release and get @krzyzacy to review the kubtest changes later, we can revisit the implementation, which looks fine to me tenatively.

We'll need to bump kubetest images to pick up these changes, I'll do that in a follow up

**Comment by krousey** [comment]
@BenTheElder Agreed. The upgrade step running in parallel is the only difference I have spotted between the downgrade suites for the release. The one that doesn't run in parallel at least gives interesting logs (and should presumably complete when it runs again in several hours).

**Comment by BenTheElder** [comment]
The image bump is ongoing, it will take a bit to build and push all of them but I can put through the config update as soon as that's done.

**Comment by BenTheElder** [comment]
see: https://github.com/kubernetes/test-infra/pull/5918 for image updates to pick this up.

File: kubetest/e2e.go [hunk 1]
```diff
@@ -164,7 +164,14 @@ func run(deploy deployer, o options) error {
 			errs = appendError(errs, err)
 		} else {
 			errs = appendError(errs, xmlWrap("UpgradeTest", func() error {
-				return skewTest(argFields(o.upgradeArgs, dump, o.clusterIPRange), "upgrade", o.checkSkew)
+				// upgrade tests really only run one spec
+				var env []string
+				for _, v := range os.Environ() {
+					if !strings.HasPrefix(v, "GINKGO_PARALLEL") {
```
Comment (krzyzacy) [inline kubetest/e2e.go]: you probably also want to handle `GINKGO_PARALLEL_NODES`

Comment (krzyzacy) [reply kubetest/e2e.go]: and ideally we want to muck with ginkgo flags rather than global envs only useful for hack/ginkgo-e2e.sh

**Comment by krousey** [comment]
Hasprefix catches that.

**Comment by krousey** [comment]
Would love to have done that, but kubetest sets the env var and also treats
them as authoritative over the flag. This was the only way.

**Comment by krzyzacy** [comment]
cc @zmerlynn 
yeah, I'd like to get rid of ginkgo-e2e.sh sometime in 2018 though :-)

**Pull request merged**